bundles/org.openhab.binding.ekey/src/main/java/org/openhab/binding/ekey/internal/handler/EkeyHandler.java

```diff
@@ -82,8 +84,8 @@ public void initialize() {
                 populateChannels(config.protocol);
                 String readerThreadName = "OH-binding-" + getThing().getUID().getAsString();
 
-                EkeyUdpPacketReceiver localReceiver = receiver = new EkeyUdpPacketReceiver(config.ipAddress,
-                        config.port, readerThreadName);
+                EkeyUdpPacketReceiver localReceiver = receiver = new EkeyUdpPacketReceiver(
+                        Optional.ofNullable(config.sourceIp).orElse(config.ipAddress), config.port, readerThreadName);
```

Comment:
Wouldn't it be better to call this natIP instead of sourceIp ? In fact, it is not the source of the packets.

Reply:
For the received packet it's the source address. I am not sure if natIP will be clearer for anyone.

Reply:
While looking on Wikipedia: https://de.wikipedia.org/wiki/Netzwerkadress%C3%BCbersetzung
This is currently the source NAT IP.

Reply:
This contradicts to your translation below (Quell IP);
Source = Quelle = ekey CV-Lan UDP converter

Reply:
I don't get it. Should I just rename it to natIp

Reply:
I don‘t know anything about kubernetes, so can’t really justify if natIP is correct, but it is definitely not the source IP.
And yes, I simply asked to rename it. For the rest, I am fine with your changes.

Reply:
I really don't know why sourceIp is not correct.
Source NAT is rewriting the source Address of a TCP/UDP Packet.
And the parameter sourceIp is the NAT rewritten sourceIp.
In my case this is e.g.
eKey Converter (x.x.x.160) -> Host (x.x.x.200) -> Kubernetes Service
The sourceIp of the packet received in the Kubernetes Service isn't x.x.x.160 but x.x.x.200.

I will update the MR later and rename it natIp.